src/main/java/com/nutomic/syncthingandroid/service/SyncthingRunnable.java

```diff
@@ -126,31 +127,15 @@ public void run() {
         try {
             if (wakeLock != null)
                 wakeLock.acquire();
+
             ProcessBuilder pb = (mUseRoot)
-                    ? new ProcessBuilder("su", "-c", TextUtils.join(" ", mCommand))
+                    ? new ProcessBuilder("su")
                     : new ProcessBuilder(mCommand);
 
-            Map<String, String> env = pb.environment();
-            // Set home directory to data folder for web GUI folder picker.
-            env.put("HOME", Environment.getExternalStorageDirectory().getAbsolutePath());
-            env.put("STTRACE", sp.getString("sttrace", ""));
-            File externalFilesDir = mContext.getExternalFilesDir(null);
-            if (externalFilesDir != null)
-                env.put("STGUIASSETS", externalFilesDir.getAbsolutePath() + "/gui");
-            env.put("STNORESTART", "1");
-            env.put("STNOUPGRADE", "1");
-            // Disable hash benchmark for faster startup.
-            // https://github.com/syncthing/syncthing/issues/4348
-            env.put("STHASHING", "minio");
-            if (sp.getBoolean("use_tor", false)) {
-                env.put("all_proxy", "socks5://localhost:9050");
-                env.put("ALL_PROXY_NO_FALLBACK", "1");
-            }
-            if (sp.getBoolean("use_legacy_hashing", false))
-                env.put("STHASHING", "standard");
-            putCustomEnvironmentVariables(env, sp);
+            HashMap<String, String> targetEnv = buildEnvironment(sp);
+            process = setupAndLaunch(pb, targetEnv);
+            targetEnv = null;
```

Comment:
There's no reason to set this to null.

Reply:
I want Java to free the memory for targetEnv since it is not needed anymore.

Reply:
That's something the compiler can figure out on its own.